### PR TITLE
Fix i18n

### PR DESCRIPTION
**Pull request by caohuilin**
# PR Details

修复 @modern-js/plugin-i18n 发布之后 export 字段导致 test 命令模块查找失败问题
修复 @modern-js/plugin-testing 类型文件
修复 tests 目录测试类型报错 

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/plugin-i18n/package.json [hunk 1]
```diff
@@ -52,6 +52,15 @@
   "publishConfig": {
     "registry": "https://registry.npmjs.org/",
     "access": "public",
-    "types": "./dist/types/index.d.ts"
+    "types": "./dist/types/index.d.ts",
+    "exports": {
```
Comment (leeight) [inline packages/cli/plugin-i18n/package.json:56]: 好像不需要把 exports 放过来吧？

File: packages/cli/plugin-testing/type.d.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import '@modern-js/core';
+import "@testing-library/react"
+import "@testing-library/jest-dom"
+import "./dist/types/runtime-testing"
+
 declare module '@modern-js/runtime/testing' {
   export * from '@testing-library/react';
-  export { renderApp, createStore } from '@modern-js/plugin-testing';
+  export { renderApp, createStore } from './dist/types/runtime-testing';
```
Comment (leeight) [inline packages/cli/plugin-testing/type.d.ts:8]: 源码里面使用 dist 感觉不太合适？
之前写 `'@modern-js/plugin-testing';` 不是挺好的么？

File: packages/cli/plugin-i18n/package.json (hunk 1 above)
Comment (caohuilin) [reply packages/cli/plugin-i18n/package.json:56]: 这里发现了一个问题，生成器的包依赖 @modern-js/codesmith-api-app，然后 @modern-js/codesmith-api-app 依赖 @modern-js/plugin-i18n，lock 文件中 @modern-js/plugin-i18n 更新最新版本时，jest.resolve.js 文件逻辑执行的时候会找不到 @modern-js/plugin-i18n 模块，需要把 exports 的 jsnext:source 去掉才能正确找到模块

File: packages/cli/plugin-testing/type.d.ts (hunk 2 above)
Comment (caohuilin) [reply packages/cli/plugin-testing/type.d.ts:8]: 直接写 @modern-js/plugin-testing 现在找不到对应的类型文件

File: packages/cli/plugin-i18n/package.json (hunk 1 above)
Comment (caohuilin) [reply packages/cli/plugin-i18n/package.json:56]: https://github.com/modern-js-dev/modern.js/issues/577

File: packages/cli/plugin-testing/type.d.ts (hunk 2 above)
Comment (leeight) [reply packages/cli/plugin-testing/type.d.ts:8]: follow up: https://github.com/modern-js-dev/modern.js/issues/577

**Review by xuchaobei — APPROVED** [review]

**Pull request merged**